### PR TITLE
logic for xtz weighted polls

### PR DESCRIPTION
**Pull request by fabiolalombardim**

### CODE REVIEW

File: utils.js [hunk 1]
```diff
@@ -117,49 +137,63 @@ const getUserTotalVotingPowerAtReferenceBlock = async (
...
+            network,
+            daoContract,
+            level
+          );
+          userVotingPower = userVotingPower.plus(userDAODepositBalance);
+        }
       }
-      userVotingPower = userVotingPower.plus(userVotePower);
+
+      return userVotingPower;
```
Comment (Man-Jain) [inline utils.js:185]: return userVotingPower - can be combined to be outside if

File: utils.js [hunk 2]
```diff
@@ -117,49 +137,63 @@ const getUserTotalVotingPowerAtReferenceBlock = async (
...
     } else {
-      const selfBalance = await getUserBalanceAtLevel(
+      let userVotingPower = new BigNumber(0);
```
Comment (Man-Jain) [inline utils.js:187]: same - can be combined to be outside if

File: components/tokens/index.js [hunk 3]
```diff
@@ -78,7 +78,16 @@ const getVotingPowerAtLevel = async (req, response) => {
...
     );
 
-    response.json({ votingWeight });
+    const votingXTZWeight = await getUserTotalVotingPowerAtReferenceBlock(
+      network,
+      address,
+      daoContract,
+      tokenID,
+      level,
+      userAddress
```
Comment (Man-Jain) [inline components/tokens/index.js:87]: shouldn't this have isXTZ = true ?

File: components/tokens/index.js [hunk 4]
```diff
@@ -78,7 +78,16 @@ const getVotingPowerAtLevel = async (req, response) => {
       userAddress
     );
 
-    response.json({ votingWeight });
+    const votingXTZWeight = await getUserTotalVotingPowerAtReferenceBlock(
```
Comment (Man-Jain) [inline components/tokens/index.js:81]: In the request we should have whether we want voting power for Xtz or Token. Don't think we should make multiple calls if not needed to avoid spamming the API and reducing load time.

File: components/tokens/index.js [hunk 5]
```diff
@@ -78,7 +78,16 @@ const getVotingPowerAtLevel = async (req, response) => {
       userAddress
```
Comment (Man-Jain) [inline components/tokens/index.js:78]: here too xtz = false would be easy to understand

File: components/choices/index.js [hunk 6]
```diff
@@ -39,10 +39,10 @@ const updateChoiceById = async (req, response) => {
   let i = 0;
 
   try {
-    const values = getInputFromSigPayload(payloadBytes);
+    const {votesData, isXTZ} = getInputFromSigPayload(payloadBytes);
```
Comment (Man-Jain) [inline components/choices/index.js:42]: Do we need to take the isXTZ here ? It's already set in the poll collection whether the poll is xtz based or not. we can take the data from there right ?

**Review by Man-Jain — CHANGES_REQUESTED** [review]

File: components/choices/index.js (hunk 6 above)
Comment (fabiolalombardim) [reply components/choices/index.js:42]: Totally right. Change made!

File: utils.js (hunk 1 above)
Comment (fabiolalombardim) [reply utils.js:185]: fixed!

File: utils.js (hunk 2 above)
Comment (fabiolalombardim) [reply utils.js:187]: fixed!

File: components/tokens/index.js (hunk 3 above)
Comment (fabiolalombardim) [reply components/tokens/index.js:87]: As specified in the task: "If isXTZ is true for any given poll, everything else remains the same, but for the vote action, the app queries a different endpoint on tzkt.io that returns the user's XTZ balance instead of the balance of the governance token."

Meaning, the user only requires to have some of the DAO's token in order to create an off-chain poll (It's XTZ balance is not important for this).

Therefore, there's no need for us to know the user's XTZ balance. If we took into consideration the user's XTZ balance in order to create an off-chain poll, any user (not a member of the DAO) would be able to create a poll for any DAO

File: components/tokens/index.js (hunk 4 above)
Comment (fabiolalombardim) [reply components/tokens/index.js:81]: We are not making multiple/useless calls. The function calls a specific API depending on whether the poll.isXTZ is set to true or false

File: components/tokens/index.js (hunk 5 above)
Comment (fabiolalombardim) [reply components/tokens/index.js:78]: added!

File: components/tokens/index.js (hunk 4 above)
Comment (fabiolalombardim) [reply components/tokens/index.js:81]: Besides, right now, this info is been useful for the frontend and I believe returning the info like this, can be benefitial for current and future tweaks/features

**Review by Man-Jain — APPROVED** [review]

File: components/choices/index.js [hunk 7]
```diff
@@ -39,10 +39,10 @@ const updateChoiceById = async (req, response) => {
   let i = 0;
 
   try {
-    const values = getInputFromSigPayload(payloadBytes);
+    const { votesData } = getInputFromSigPayload(payloadBytes);
```
Comment (Man-Jain) [inline components/choices/index.js:42]: Can you change this to what it was earlier so we don't do unnecessary changes in the code. I feel like this might break the code since there's a slight diffence in `const value` and `const { votesData}`

File: components/tokens/index.js (hunk 3 above)
Comment (Man-Jain) [reply components/tokens/index.js]: Okay, If that's the case it should be fine. Do check though if this function is being used in other places don't cause any side effects on the frontend.

File: components/choices/index.js (hunk 7 above)
Comment (fabiolalombardim) [reply components/choices/index.js:42]: fixed!
